src/Kernel32.Desktop/Kernel32.Helpers.cs

```diff
@@ -197,26 +197,27 @@ public static IEnumerable<MODULEENTRY32> Module32Enumerate(SafeObjectHandle hSna
             }
         }
 
-        public static string QueryFullProcessImageName(
+        public static unsafe string QueryFullProcessImageName(
             SafeObjectHandle hProcess,
             QueryFullProcessImageNameFlags dwFlags = QueryFullProcessImageNameFlags.None)
         {
             // If we ever resize over this value something got really wrong
             const int maximumRealisticSize = 1 * 1024 * 2014;
 
-            var buffer = new StringBuilder(255);
+            int size = 255;
+
             do
             {
-                int size = buffer.Capacity;
-                bool success = QueryFullProcessImageName(hProcess, dwFlags, buffer, ref size);
+                var text = stackalloc char[size];
```

Comment:
2 remarks here :

1. Beware of the memory usage of stackallock, as memory is only freed when the stack is unwind (end of function) this while loop uses 3.5 Mo instead of 2 Mo max as evoked by `maximumRealisticSize`
2. The default thread stack size limit is often 1 Mo so this code will stackoverflow before `maximumRealisticSize` anyway.

Better use normal (heap) allocations here.